syft/frameworks/torch/nn/functional.py

```diff
@@ -41,6 +41,32 @@ def dropout(input, p=0.5, training=True, inplace=False):
     return input
 
 
+results = {}
+
+
+def inverse(var):
```

Comment:
MAYBE: We have this in ```precision.py```
```
def reciprocal(self, method="NR", nr_iters=10):

Args:
             method:
             'NR' : `Newton-Raphson`_ method computes the reciprocal using iterations
                     of :math:`x_{i+1} = (2x_i - self * x_i^2)` and uses
                     :math:`3*exp(-(x-.5)) + 0.003` as an initial guess by default
         
             nr_iters:
                 Number of iterations for `Newton-Raphson`
         Returns:
             Reciprocal of `self`
         """
 
         if method.lower() == "nr":
             new_self = self.modulus()
             result = 3 * (0.5 - new_self).exp() + 0.003
             for i in range(nr_iters):
                 result = 2 * result - result * result * new_self
             return result * self.signum()
```

Are the results better/computed faster if you use this?

Reply:
Interesting, I should compare!
But the key the fast computtaion here is caching the result, for batchnorm at inference that's always the inverse you're trying to do :)

Reply:
Might be worth checking the following:
* if changing the initial guess gives better results? (like the difference between the logits would be smaller)
* if removing the if changes the speed -- sometimes computers do this ```branch prediction``` thing.
CPUs are using a pipeline for fetching data/fetching instructions/etc and they might run code before it needed to be run (predict that you might take the "if" or "else" branch). If the prediction is correct you get "full speed" ahead, but the problem comes in when that prediction is bad, in which case they need to evict all the results from the pipeline (and if this is the case) it might introduce some latency.

I do not know all the "internals" of python to know what it does with the ```if....else```, but it might be worth a shot removing the ```if...else``` and check if it speeds things up

Reply:
Integrated :) 
I'll keep the if else statement for the moment because this function isn't called so many times
I could improve the initiatal guess indeed for my method, i'll investigate this separateely I think